### PR TITLE
Remove Applicative instance of Fold and release teeWith

### LINKED ISSUE #499: Applicative instance of Fold and the new Tee type
See also #455 .

Since we are now including terminating/splitting functionality under the 'Fold' type we need to reconsider the existing applicative instance. The current applicative instance is consistent with the `foldl` library. However, with terminating folds a splitting applicative like the `Parser` type is possible for `Fold`. 

To keep Folds consistent with Parsers we can change the applicative instance of folds to a splitting applicative. The current applicative would become a `teeWith` function like in Parsers. This would be a breaking change. To mitigate the impact of the change to existing users we can have a new `Tee` type whose applicative would behave just like the current Fold applicative. This would be similar to the way we have `ZipSerial` type in streams.

### PR DESCRIPTION
**Pull request by adithyaov**
Closes #499

### CODE REVIEW

File: Changelog.md [hunk 1]
```diff
@@ -2,8 +2,8 @@
 
 ### Notice
 
-* In the next major release the behaviour of `<*>` for folds will be updated
-  from `teeWith` to `splitWith` to reflect it's similarity with parsers.
+* In the next major release `Applicative` for folds will be re-added and the
```
Comment (harendra-kumar) [inline Changelog.md:5]: We can remove the notice.

File: Changelog.md [hunk 2]
```diff
@@ -70,6 +71,8 @@
     * `mapM` to `rmapM`
     * `sequence` to `rsequence`
 
+* Remove `Applicative` instance of folds.
```
Comment (harendra-kumar) [inline Changelog.md:74]: This should go in breaking changes, not in deprecations.

Comment (harendra-kumar) [reply Changelog.md:74]: Also, add "Please use `teeWith` or the `Tee` type as an alternative to Fold applicative.".

**Review by harendra-kumar — CHANGES_REQUESTED** [review]
Also remove the `Semigroup`, `Monoid`, `Num`, `Fractional`, `Floating` instances. They belong to the `Tee` type. For Fold type if we want to have those then they should agree with the Fold applicative.

**Comment by adithyaov** [comment]
> Also remove the Semigroup, Monoid, Num, Fractional, Floating instances. They belong to the Tee type. For Fold type if we want to have those then they should agree with the Fold applicative.

Yeah was about to ask you the behaviour.

**Review by harendra-kumar — APPROVED** [review]

**Comment by adithyaov** [comment]
Closes #499

**Pull request merged**